crawler/src/response.rs

```diff
@@ -0,0 +1,46 @@
+use std::sync::mpsc;
+
+use data::UrlMsg;
+
+struct Response {
+    status: usize,
+    headers: Vec<String>,
+    sender: mpsc::Sender<UrlMsg>,
+    url: String
+}
+
+impl Response {
+    fn new(status: usize,
+           headers: Vec<String>,
+           sender: mpsc::Sender<UrlMsg>,
+           url: String) -> Response {
+        Response {
+            status: status,
+            headers: headers,
+            sender: sender,
+            url: url
+        }
+    }
+
+    fn respond(&self) {
+        let mut new_urls = Vec::new();
+        match self.status {
+            200 => new_urls.push_all(&parse_ok(&self.headers)),
+            301 => new_urls.push(parse_moved(&self.headers)),
+            403 => (),
+            500 => new_urls.push(self.url.clone()),
+            _ => unreachable!()
+        }
```

Comment:
If i'm gonna return an empty vec there's probably a better way of doing this right? Optional...?